Makefile

```diff
@@ -34,7 +34,57 @@ endif
 GO=go
 GOFLAGS=""
 
+# requires:
+#	AWS_SECRET		- Secret from AWS
+#	AWS_ACCESS		- Access key from AWS
+#	S3_REGION		- Region of the bucket used to store the model
+#	S3_ENDPOINT		- Endpint of the bucket
+#	REGISTRY_USERNAME	- quay.io username
+#	REGISTRY_PASSWORD	- quay.io password
+#	REGISTRY_CREDENTIALS	- quay.io robot secret file
```

Comment:
To be honest this seems to be used just in `oc create -f ${REGISTRY_CREDENTIALS}` but that would create unsubstituted secret.

Reply:
> It seems odd to require username and password, and also separately a secret file for "credentials"

True, the credentials file is for the quay robot you use to push to your repo. Either way we would need to set them in the file so I think we need to think about these sepretely at some point anyway, either as test arguments or before when setting up the credentials file

> To be honest this seems to be used just in oc create -f ${REGISTRY_CREDENTIALS} but that would create unsubstituted secret

I am not really sure what this means, should I not be doing it this way? @MarianMacik

Reply:
My comment was that 2 secrets are created for image registry credentials but just one is really used - the one linked to the pipeline service account. In other words, the second sed is not needed AFAIK.

Reply:
The `IMAGE_REGISTRY_CREDENTIALS` file is only one secret and that is linked. I am not sure which secret you are referring to. I got the linking step from the README so maybe that is not needed actually? Or are you saying the username and password secrets are not required?

Reply:
I think what @MarianMacik is saying is that it's not clear why we need to create two secrets in these two lines in the `go-test-setup`:
```
@sed -e "s#{{ IMAGE_REGISTRY_USERNAME }}#${IMAGE_REGISTRY_USERNAME}#g" -e "s#{{ IMAGE_REGISTRY_PASSWORD }}#${IMAGE_REGISTRY_PASSWORD}#g" pipelines/tekton/aiedge-e2e/templates/credentials-image-registry.secret.yaml.template | oc create -f -
oc create -f ${IMAGE_REGISTRY_CREDENTIALS}
```

The first one is going to create a secret called `credentials-image-registry` based on the referenced template file, and the second is whatever is in the file referenced by `IMAGE_REGISTRY_CREDENTIALS`. Could/should both contain details to the same account? E.g. should `IMAGE_REGISTRY_USERNAME` and `IMAGE_REGISTRY_PASSWORD` be the quay robot account name and token? In which case, should there just be one secret? Or is there an expectation that they're accounts? Potentially different registries?

Reply:
I think the biggest reason why these are not the same is because the `IMAGE_REGISTRY_CREDENTINALS` is gotten from quay itself as its own yaml file they provide, this will be unique for every user's account. The password and username template is fine as that is all we need to change for it, so it is not actually about the file.

But for the other one the file is what is actually significant, not sure if my thoughts made sense there

Reply:
From https://github.com/opendatahub-io/ai-edge/tree/main/pipelines#setup-credentials:

> Image Registry - [credentials-image-registry.secret.yaml.template](https://github.com/opendatahub-io/ai-edge/blob/main/pipelines/tekton/aiedge-e2e/templates/credentials-image-registry.secret.yaml.template) to include the username and password with write access to the image repository. This is needed only in case you noted username and password of the robot account. In case you directly downloaded and applied a Kubernetes Secret, this file can be skipped.

So it sounds like _*either*_ you need to have a `.dockerconfigjson` secret file, similar to the one that you can get from a Quay robot account (there are a bunch of things you can get from a robot account, including username/password), and have that linked to the SA; _*or*_ create a secret from the template by substituting the template values and then linking that to the SA. Does that sound right?

The problem with this `go-test-setup` make target recipe, is that it's requiring both, and only linking the `IMAGE_REGISTRY_CREDENTIALS` one. If only one is required, for simplicity let's just use the simplest one (probably the robot file one).

WDYT?

Reply:
> In case you directly downloaded and applied a Kubernetes Secret, this file can be skipped.

Agh okay I guess I missed this part when reading, now I understand the comments from before.

Reply:
I think I would actually rather the username and password because then there is less manual work for running the tests, and having an extra file laying around is annoying. But maybe that isn't the only consideration.

Reply:
Sure, I'm easy with either :+1: